api/src/utils/media/csv/validation/csv-row-validator.ts

```diff
@@ -352,3 +352,55 @@ export const getValidFormatFieldsValidator = (config?: ColumnFormatValidatorConf
     return csvWorksheet;
   };
 };
+
+export type FileColumnUniqueValidatorConfig = {
+  file_column_unique_validator: {
+    key_columns: string[];
+  };
+};
+
+export const getUniqueColumnsValidator = (config?: FileColumnUniqueValidatorConfig): CSVValidator => {
+  return (csvWorksheet) => {
+    if (!config) {
+      return csvWorksheet;
+    }
+
+    if (config.file_column_unique_validator.key_columns.length < 1) {
+      return csvWorksheet;
+    }
+
+    const keySet = new Set();
+    const rows = csvWorksheet.getRowObjects();
+    const lowercaseHeaders = csvWorksheet.getHeadersLowerCase();
+
+    const columnIndices = config.file_column_unique_validator.key_columns.map((column) =>
+      lowercaseHeaders.indexOf(column.toLocaleLowerCase())
+    );
+    if (columnIndices.indexOf(-1) !== -1) {
```

Comment:
Is there a more clear or obvious check that could assert that the worksheet or config columns don't line up? If not, maybe a comment could clarify what is happening here?

Reply:
I've added a comment that hopefully makes more sense. The check was done in this way because there can be multiple columns provided, so I was checking for a not found instead of looping through. Let me know if it makes more sense